extensions/wikia/TemplateDraft/TemplateDraftHooks.class.php

```diff
@@ -76,24 +77,51 @@ public static function onEditFormPreloadText( &$text, Title $title ) {
 	 * Triggered if a user edits a Draft subpage of a template.
 	 * It adds an editintro message with help and links.
 	 *
-	 * @param String $msgName
-	 * @param Array $msgParams
+	 * @param Array $preloads
 	 * @param Title $title
 	 * @return bool
 	 */
-	public static function onEditPageLayoutShowIntro( &$msgName, &$msgParams, Title $title ) {
+	public static function onEditPageLayoutShowIntro( &$preloads, Title $title ) {
 		$helper = new TemplateDraftHelper();
-
-		if ( $helper->isTitleNewDraft( $title )
-			&& class_exists( 'TemplateConverter' )
-			&& TemplateConverter::isConversion()
-		) {
-			$msgName = 'templatedraft-editintro';
-
-			$base = Title::newFromText( $title->getBaseText(), NS_TEMPLATE );
-			$msgParams = [ $base->getFullUrl( ['action' => 'edit'] ) ];
+		if ( $title->getNamespace() == NS_TEMPLATE ) {
+			if ( $helper->isTitleDraft( $title )
+				&& class_exists( 'TemplateConverter' )
+				&& TemplateConverter::isConversion()
+			) {
+				$base = Title::newFromText( $title->getBaseText(), NS_TEMPLATE );
+				$baseHelp = Title::newFromText( 'Help:PortableInfoboxes' );
+				$preloads['EditPageIntro'] = [
+					'content' => wfMessage( 'templatedraft-editintro' )->rawParams(
+						Xml::element( 'a', [
+							'href' => $baseHelp->getFullURL(),
+							'target' => '_blank',
+						],
+							wfMessage( 'templatedraft-module-help' )
+						),
+						Xml::element( 'a', [
+							'href' => $base->getFullUrl( [ 'action' => 'edit' ] ),
+							'target' => '_blank'
+						],
+							wfMessage( 'templatedraft-module-view-parent' ) )
+					)->escaped(),
+				];
+			} elseif ( !$helper->isTitleDraft( $title ) ) {
+				$base = Title::newFromText( $title->getBaseText() . '/Draft', NS_TEMPLATE );
+				$draftUrl = $base->getFullUrl( [
+					'action' => 'edit',
+					TemplateConverter::CONVERSION_MARKER => 1,
+				] );
+				$preloads['EditPageIntro'] = [
+					'content' => wfMessage( 'templatedraft-module-editintro-please-convert' )->rawParams(
+						Xml::element( 'a', [
+							'href' => $draftUrl,
+							'target' => '_blank'
+						],
+							wfMessage( 'templatedraft-module-button' ) )
```

Comment:
Should use an explicit output format. Right now, without one, it will default to `->parse()` which is probably unnecessary given the message text, and the contents of `Xml::element` are escaped anyway, so `->plain()` or `->text()` would be fine here. :)